tools/shared.py

```diff
@@ -2979,8 +2979,20 @@ def delebify(buf, offset):
 
   @staticmethod
   def add_emscripten_metadata(js_file, wasm_file):
-    mem_size = Settings.STATIC_BUMP
+    WASM_PAGE_SIZE = 65536
+
+    mem_size = Settings.TOTAL_MEMORY // WASM_PAGE_SIZE
     table_size = Settings.WASM_TABLE_SIZE
+    global_base = Settings.GLOBAL_BASE
+
+    js = open(js_file).read()
+    m = re.search(r"tempDoublePtr\s+=\s+(\d+)", js)
+    tempdouble_ptr = int(m.group(1))
+    m = re.search(r"DYNAMIC_BASE\s+=\s+(\d+)", js)
+    dynamic_base = int(m.group(1))
+    m = re.search(r"DYNAMICTOP_PTR\s+=\s+(\d+)", js)
+    dynamictop_ptr = int(m.group(1))
```

Comment:
This seems strangely convoluted to read back in the file we just generated.

Reply:
It would require more extensive changes to force the JS compiler to use a value for `DYNAMICTOP_PTR` that would be chosen by Python. Right now it's automatically allocated by `makeStaticAlloc()`. I thought the method of parsing the JS was acceptable since it was done in earlier incarnations of this function.